packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/PdfViewerHighlight.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { FC, useMemo, useEffect } from 'react';
+import cx from 'classnames';
+import { settings } from 'carbon-components';
+import { QueryResult } from 'ibm-watson/discovery/v2';
+import { ProcessedDoc } from 'utils/document';
+import { TextMappings } from '../../types';
+import { PdfDisplayProps } from '../PdfViewer/types';
+import { PdfRenderedText } from '../PdfViewer/PdfViewerTextLayer';
+import { DocumentFieldHighlight } from './types';
+import { ExtractedDocumentInfo } from './utils/common/documentUtils';
+import { Highlighter } from './utils/Highlighter';
+
+type Props = PdfDisplayProps & {
+  /**
+   * Class name to style highlight layer
+   */
+  className?: string;
+
+  /**
+   * Class name to style each highlight
+   */
+  highlightClassName?: string;
+
+  /**
+   * Document data returned by query
+   */
+  document: QueryResult;
+
+  /**
+   * Parsed document information
+   */
+  parsedDocument: ExtractedDocumentInfo | null;
+
+  /**
+   * Highlight spans on fields in document
+   */
+  highlights: DocumentFieldHighlight[];
+
+  /**
+   * PDF text content information in a page from parsed PDF
+   */
+  pdfRenderedText: PdfRenderedText | null;
+
+  /**
+   * Flag to whether or not to use bbox information from html field in the document.
+   * True by default. This is for testing and debugging purpose.
```

Comment:
I think that we should better denote that these properties aren't meant to be used normally by end users. Best I can think is to prepend an underscore to the name:

```js
  _useHtmlBbox?: boolean;
```